pkg/util/inplaceupdate/inplace_update_defaults.go

```diff
@@ -135,15 +140,46 @@ func defaultPatchUpdateSpecToPod(pod *v1.Pod, spec *UpdateSpec, state *appspub.I
 			state.NextContainerImages[c.Name] = newImage
 		}
 	}
+
+	containersResourceChanged := sets.NewString()
```

Comment:
it seems that containersResourceChanged is not used at all. consider replace containersResourceChanged with a map from container name to ResourceRequirements. In this way, one can skip UpdateResource in L153, and call UpdateResource in L180 in one shot